twitter/src/main/java/com/hazelcast/jet/contrib/twitter/TwitterSources.java

```diff
@@ -0,0 +1,204 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.contrib.twitter;
+
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.internal.json.Json;
+import com.hazelcast.internal.json.JsonObject;
+import com.hazelcast.jet.pipeline.SourceBuilder;
+import com.hazelcast.jet.pipeline.StreamSource;
+
+import com.twitter.hbc.ClientBuilder;
+import com.twitter.hbc.core.endpoint.StreamingEndpoint;
+import com.twitter.hbc.httpclient.BasicClient;
+import com.twitter.hbc.core.processor.StringDelimitedProcessor;
+import com.twitter.hbc.httpclient.auth.Authentication;
+import com.twitter.hbc.httpclient.auth.OAuth1;
+
+import java.util.ArrayList;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.Properties;
+
+import javax.annotation.Nonnull;
+
+
+/**
+ * Contains methods for creating Twitter stream sources.
+ */
+public final class TwitterSources {
+
+    private TwitterSources() {
+    }
+
+    /**
+     * Creates a {@link StreamSource} which reads tweets from Twitter's Streaming API
+     * for data ingestion to Jet pipelines.
+     *
+     * Example usage:
+     * <pre>{@code
+     * Properties credentials = loadTwitterCredentials();
+     * List<String> terms = new ArrayList<String>(Arrays.asList("BTC", "ETH"));
+     * StreamSource<String> streamSource =
+     *              TwitterSources.stream(
+     *                      "twitter-stream-source",
+     *                      () -> new StatusesFilterEndpoint().trackTerms(terms),
+     *                      credentials,
+     *                      Constants.STREAM_HOST
+     *              );
+     * Pipeline p = Pipeline.create();
+     * StreamSourceStage<String> srcStage = p.readFrom(streamSource);
+     * }</pre>
+     *
+     * @param name             a descriptive name of this source.
+     * @param endpointSupplier a supplier function that supplies a Twitter StreamingEndpoint to connect to source.
+     * @param credentials      a Twitter OAuth1 credentials that consists "consumerKey",
+     *                         "consumerSecret", "token", "tokenSecret" keys.
+     * @param host             a Twitter host URL to connect.
+     *                         These hosts are defined in {@link com.twitter.hbc.core.Constants}.
+     * @return a stream source to use in {@link com.hazelcast.jet.pipeline.Pipeline#readFrom}
+     */
+    @Nonnull
+    public static StreamSource<String> stream(@Nonnull String name,
```

Comment:
I think user doesn't need to supply a name for the source. We can provide a generic name for the source.

Reply:
I removed the source names from method signatures